controllers/profile.js

```diff
@@ -1,14 +1,23 @@
 // const User = require("../models/user");
 const Post = require("../models/post");
+const User = require("../models/user")
 
 const ProfileController = {
   Index: (req, res) => {
     const userObject = req.session.user;
+    const usersFriends = [];
+
+    userObject.friends.forEach(friend => {
+      User.find({ _id: friend }).then(user => {
+        usersFriends.push(user[0]);
+      });
+    });
 
     Post.find({ user: userObject._id })
     .populate("user")
     .then((posts) => {
-      res.render("profile/index", { user: userObject, posts: posts });
+      console.log("CHECKING ADDITION OF USER: ", usersFriends);
+      res.render("profile/index", { user: userObject, posts: posts, friends: usersFriends });
```

Comment:
could this sometimes not work if this takes less time to complete that the above find for usersFriends?

Reply:
Maybe, I'm not 100% sure how the event loop functions in this situation. This is all being refactored anyway which is why I left it in, we're completely changing the way this controller functions, just need this merge to start working on that, otherwise it bites off too much for this branch.